libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.java

```diff
@@ -76,8 +77,26 @@ static Builder builder(Context context, LocaleUtils localeUtils) {
    * @param callback a RetroFit callback which contains an onResponse and onFailure
    * @since 0.5.0
    */
-  public void getRoute(Callback<DirectionsResponse> callback) {
-    mapboxDirections.enqueueCall(callback);
+  public void getRoute(final Callback<DirectionsResponse> callback) {
+    final RouteRetrievalInfo.Builder routeRetrievalBuilder = RouteRetrievalInfo.builder().start();
+    mapboxDirections.enqueueCall(new Callback<DirectionsResponse>() {
+      @Override
+      public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
+        routeRetrievalBuilder.end();
+        callback.onResponse(call, response);
+        NavigationTelemetry.getInstance().routeRetrievalEvent(
+          routeRetrievalBuilder
+            .route(response.body().routes().get(0))
```

Comment:
Does this route retrieval event only consider the first route returned in a directions API response?  If so, we should check the response is successful and has a valid list of route before getting `0` - possible `IndexOutOfBounds` exception.